src/plugins/es_ui_shared/docs/docs/form_lib/core/form_hook.md

```diff
@@ -0,0 +1,208 @@
+  const { form } = useForm();
+  
+  const onClickSubmit = async () => {
+    // We validate all the form fields and set the "isValid" state to true or false
+    const { isValid, data } await form.submit();
+  
+    if (isValid) {
+      // ...
+    }
+  };
+  
+  return (
+    <Form form={form}>
+      ...
+      <button onClick={onClickSubmit}>Submit</button>
+      {form.isValid === false && (
+        <div>Only show this message if the form validity is "false".</div>
+      )}
+    </Form>
+  );
+}
+```
+
+* calling the `validate()` handler on the form. As you can see in the example below, if you don't use the `form.submit()`, you have to manually declare and update the `isSubmitting` and `isSubmitted` states.
+
+```js
+export const MyComponent = ({ onFormUpdate }: Props) => {
+  const [isSubmitted, setIsSubmitted] = useState(false);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const { form } = useForm<UserFormData>();
+
+  const onClickSubmit = async () => {
+    setIsSubmitted(true);
+    setIsSubmitting(true);
+
+    // If the "isValid" state is "undefined" (=== not all the fields are dirty),
+    // call validate() to run validation on all the fields.
+    const isValid = form.isValid ?? (await form.validate());
+    setIsSubmitting(false);
+
+    if (isValid) {
+      console.log('Form data:', form.getFormData());
+    }
+  };
+
+  const hasErrors = isSubmitted && form.isValid === false;
+
+  return (
+    <Form form={form}>
+      <UseField path="firstName" config={{ validations: [{ validator }] }} />
+      <UseField path="lastName" />
+
+      <button
+        onClick={onClickSubmit}
+        disabled={isSubmitting || hasErrors}
+      >
+        {isSubmitting ? 'Sending...' : 'Submit'}
+      </button>
+      {hasErrors && <div>Form is invalid.</div>}
+    </Form>
+  );
+};
+```
+
+### id
+
+**Type:** `string`
+
+The form id. If none was provided, "default" will be returned.
+
+## Handlers
+
+### submit()
+
+**Returns:** `Promise<{ data: T | {}, isValid: boolean }>`
+
+This handlers submits the form and returns its data and validity. If the form is not valid, the data will be `null` as only valid data is passed through the `serializer(s)` before being returned.
+
+```js
+const { data, isValid } = await form.submit();
+```
+
+### validate()
+
+**Returns:** `Promise<boolean>`
+
+Use this handler to get the validity of the form.
+
+```js
+const isFormValid = await form.validate();
+```
+
+### getFields()
+
+**Returns:** `{ [path: string]: FieldHook }`
+
+Access any field on the form.
+
+```js
+const { name: nameField }  = form.getFields();
+```
+
+### getFormData()
+
+**Arguments:** `options?: { unflatten?: boolean }`  
+**Returns:** `T | R`
+
+Return the form data. It accepts an optional options with an `unflatten` parameter (defaults to `true`). If you are only interested in the raw form data, pass `unflatten: false` to the handler ([read more about the "Raw" data state here](in_out_raw_state.md#raw-data-state)).
```

Comment:
```suggestion
Return the form data. Accepts an optional `options` with an `unflatten` parameter (defaults to `true`). If you are only interested in the raw form data, pass `unflatten: false` to the handler ([read more about the "Raw" data state here](in_out_raw_state.md#raw-data-state)).
```

Reply:
I guess this is going away after: https://github.com/elastic/kibana/issues/78589 ?

Might be good to make a not of that as part of the work to be done along with that PR change (i.e., that the docs need to be updated)

Reply:
Good point, I will add a note for it in the PR 👍